pennylane/ops/functions/equal.py

```diff
@@ -658,41 +658,54 @@ def _equal_measurements(
     """Determine whether two MeasurementProcess objects are equal"""
 
     if op1.obs is not None and op2.obs is not None:
-        return equal(
+        dispatch_result = _equal(
             op1.obs,
             op2.obs,
             check_interface=check_interface,
             check_trainability=check_trainability,
             rtol=rtol,
             atol=atol,
         )
+        if isinstance(dispatch_result, str):
+            return dispatch_result
+        return dispatch_result
 
     if op1.mv is not None and op2.mv is not None:
         if isinstance(op1.mv, MeasurementValue) and isinstance(op2.mv, MeasurementValue):
-            return qml.equal(op1.mv, op2.mv)
+            mv_equal = qml.equal(op1.mv, op2.mv)
+            if isinstance(mv_equal, str):
+                return mv_equal
+            return mv_equal
```

Comment:
So we know that `equal` itself always returns a boolean, so we would never hit the `str` branch.

I don't think we need to add too detailed of a message if the measurement values are not the same, but we should probably still return a message like `op1 and op2 are not equal because they have different measurement values` in the case that the measurement values are not equal.